### DISCUSSION TITLE
Import firebase into component, break storybook

### DISCUSSION THREAD

**Discussion by TadeuA** [storybookjs/storybook#16502] [Help]
If you have a firebase service import in the component, its story is not rendered.
If removing the import the story works normally

My component is like this

```
//...
import {fireLogEvent} from 'services/firebase';

export const Button: FC<T.IButton> = (props) => {
  const handleClick = useCallback((event) =>{
   !!props.logEvent && fireLogEvent(props.logEvent.logName, props.logEvent.params);
    props.onClick(event);
  },[props.onClick, props.logEvent])
//...
```

That way it doesn't render the story.

Now just comment the lines

```
//...
//import {fireLogEvent} from 'services/firebase';

export const Button: FC<T.IButton> = (props) => {
  const handleClick = useCallback((event) =>{
   //!!props.logEvent && fireLogEvent(props.logEvent.logName, props.logEvent.params);
    props.onClick(event);
  },[props.onClick, props.logEvent])
//...

```
And it works normally.

I would like the story to work without having to comment the lines, otherwise this is not functional for the project.

**Comment by kws1207** [comment]
Any updates?

**Comment by moishinetzer** [comment]
I believe this is an issue with the way storybook handles tree-shaking. Naturally, storybook can't handle server-side code so for some reason whenever there is a specific server import storybook breaks. 

In fact, even if you import a file that imports firebase but doesn't even use it in that file storybook breaks as well! 

I had this exact issue but with a component that imported my prisma client and didn't use it. Reading the devtools console will help finding the source of which file is importing the server-side code.

Can we investigate the way tree-shaking works properly? I wonder if storybook checks all the server-side code BEFORE tree-shaking regardless.
